kmath-units/src/commonMain/kotlin/space/kscience/kmath/units/units.kt

```diff
@@ -0,0 +1,276 @@
+public fun T(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e12)
+public fun G(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e9)
+public fun M(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e6)
+public fun k(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e3)
+public fun h(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e2)
+public fun da(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e1)
+
+public fun y(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-24)
+public fun z(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-21)
+public fun a(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-18)
+public fun f(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-15)
+public fun p(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-12)
+public fun n(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-9)
+public fun u(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-6)
+public fun m(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-3)
+public fun c(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-2)
+public fun d(measure: Measure): Measure = measure.copy(multiplier = measure.multiplier * 1e-1)
+
+public infix fun Measure.pow(power: Int): Measure =
+    copy(chain = chain.mapValues { (_, v) -> IntRing.power(v, power.toUInt()) }, multiplier = multiplier.pow(power))
+
+public open class MeasurementAlgebra<T>(public open val algebra: Algebra<T>) : Algebra<Measurement<T>> {
+    public operator fun T.times(m: Measure): Measurement<T> = Measurement(m, this)
+    public operator fun Measurement<T>.times(m: Measure): Measurement<T> =
+        copy(measure = MeasureAlgebra { measure * m })
+
+    public override fun bindSymbol(value: String): Measurement<T> = algebra.bindSymbol(value) * pure
+}
+
+public fun <T> Algebra<T>.measurement(): MeasurementAlgebra<T> = MeasurementAlgebra(this)
+
+public open class MeasurementSpace<T, out A>(public override val algebra: A) : MeasurementAlgebra<T>(algebra),
+    Group<Measurement<T>>, ScaleOperations<Measurement<T>> where A : Group<T>, A : ScaleOperations<T> {
+    public override val zero: Measurement<T>
+        get() = Measurement(pure, algebra.zero)
+
+    public override fun add(a: Measurement<T>, b: Measurement<T>): Measurement<T> {
+        require(a.measure.chain == b.measure.chain) {
+            "The units are incompatible. The chains are ${a.measure.chain} and ${b.measure.chain}"
+        }
+
+        return a.copy(value = algebra { a.value * a.measure.multiplier + b.value * b.measure.multiplier })
```

Comment:
I think code should convert `b.value` from scale used in `b` to scale used in `a` so it be something like `a.value + b.value * b.measure.multiplier / a.measure.multiplier`?

test cases:
- `(2 * gramm) + (0 * gramm)` should be equals to `(2 * gramm)`
- addition of meters and foots